### ISSUE TITLE
full screen presentation on second display

### ISSUE THREAD

**Issue by hiroshisuga** [bigbluebutton/bigbluebutton#11403]
**Is your feature request related to a problem? Please describe.**
My university is moving to the "hybrid lecture", which is to broadcast the face-to-face lecture online. To do so, I need to do a BBB meeting in the full-screen mode and show it to students by a projector in a lecture room, while other students watch it online. However, the chat and user panes are invisible during the full screen mode.

**Describe the solution you'd like**
I would like to have an option to drag (or copy) the presentation pane on the second display (which is the projector) and show it full-screen, while the other panes (such as chat, user list, and other controllers) remain available in the main display (which is the PC's display). 

**Describe alternatives you've considered**
A magnifier app (copying the presentation area to the app window) would be an acceptable option. 

As far as I read the documents, JS does not have an ability to show a full-screen content on the second display, so opening a new window to show the presentation container (and the presenter can move to the second screen and full-screened) might do the trick.

**Additional context**
In Japanese universities, doing lectures online in parallel to face-to-face (my university calls it hybrid lecture) seems to be on an upward trend from the coming spring.

**Comment by pedrobmarin** [comment]
Hi @hiroshisuga 

Although I believe the multiple clients idea is the easiest way to achieve this, I already saw some folks using Portals (https://reactjs.org/docs/portals.html) to display react components in different browser tabs/windows.

**Comment by hiroshisuga** [comment]
Hi @pedrobmarin ,
Thanks for the suggestion. I was not aware of Portals. Would this not make a time lag between two clients in different tabs? 
I was simply trying to launch one more BBB client (so I had a presenter BBB and a viewer BBB in one PC) and show the viewer's screen on the projector. This however makes an unignorable time lag in the face-to-face lecture.

**Comment by hiroshisuga** [comment]
As far as I read the document, Portals provides a way to "move" a whole DOM to another DOM hierarchy. Not "copy". That appears to be a great solution. Thanks! 
BTW, is this (#8046) the one you mean?

**Comment by pedrobmarin** [comment]
Nah, it's from another project that they used to make a custom Picture-In-Picture for multiple videos. It's called eduMEET (https://edumeet.org)

**Comment by markginter31** [comment]
I use a chromecast device connected to the projector and then cast the tab of the bbb session to the chromecast.  I could see where you would open two tabs and log into both and then cast the one tab and then fullscreen it....

**Comment by hiroshisuga** [comment]
This is what I have now. Still a lot of work need to be done. But looks good!
If the community is interested in this function, I would work more and send a PR.
![screen](https://user-images.githubusercontent.com/45039819/109253917-35223980-7834-11eb-8828-a7e718affd4c.gif)

**Comment by ant0nwax** [comment]
+1

**Comment by gustavotrott** [comment]
![image](https://github.com/bigbluebutton/bigbluebutton/assets/5660191/7ddc2e3c-eb45-457d-8c65-52065e3d6391)

**Comment by MairusuPawa** [comment]
Hello! Just showing here interest for this feature as well. I've noticed work was apparently started on that feature, but went stall. Are there any updates I'm not aware of? Thanks!

**Comment by hiroshisuga** [comment]
Hi @MairusuPawa, after the start of v3.0, the discussion stalled. I was expecting a better implementation using the plug-in system. But it does not seem to allow exporting the whole presentation yet.
If you are interested, I am developing a private PR to allow detaching the presentation container, which you can move to the second monitor and show on fullscreen.
https://github.com/hiroshisuga/bigbluebutton/pull/240
I would be grateful if you report any problem or suggestion.

**Comment by gustavotrott** [comment]
I believe the plugin system can handle this now.

You can duplicate the user session while forcing a specific layout in the new session:

```js
const joinUrl = await pluginApi.getJoinUrl({
  enforceLayout: 'PRESENTATION_ONLY',
});
```

With the `joinUrl`, you can simply open a new window using that URL.

I have a plugin with a similar idea. It is not fully ready yet, but it can serve as inspiration:

https://github.com/gustavotrott/bbb-plugin-remote-camera


Using this idea the presentation would keep displaying in both windows. I think it is good for presenters.

**Comment by hiroshisuga** [comment]
Hi @gustavotrott , thanks for the idea. I am looking into the plugin system more closely now. 
By simply replacing CAMERA_ONLY with PRESENTATION_ONLY, I can duplicate the presentation window. No need to modify the source code. This would be definitely useful for teachers or professors.
But unfortunately there is a little delay in refreshing the separated presentation, similar to the case where the presenter logs-in using another client to show the presentation on the projector.
When teaching, this delay frustrates the presenter, especially when the network is slow. A solution to duplicate the presentation without transmitting the data through network is preferable.

**Comment by gustavotrott** [comment]
Using the presentation on one monitor and a second session on another will inevitably introduce network delay.

The only way to avoid that delay is to run the whiteboard directly on the fullscreen monitor and simply ignore or hide the presentation on the desk monitor.

The plugin system does provide a way to access the current slide with annotations (which could feed a pop-up), but it does not include the mouse cursor.

A possible workaround would be to use three monitors:

* One for the desk with chat and user list
* A second for the desk with the presentation (via a duplicated session using `PRESENTATION_ONLY`)
* A third (projector) mirroring the second monitor for the students

I do not see a better solution at the moment.

**Comment by gustavotrott** [comment]
@hiroshisuga feel free to share any ideas on how the plugin system could be improved to better support your use case. We can try to do it!
